modules/serverless-logic-security-example-oauth-authentication.adoc

```diff
@@ -0,0 +1,111 @@
+          }
+        ]
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "oauth-example": {
+        "type": "oauth2",
+        "flows": {
+          "clientCredentials": {
+            "authorizationUrl": "https://example.com/oauth",
+            "tokenUrl": "https://example.com/oauth/token",
+            "scopes": {}
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+In this example, the `sayHelloOauth2` operation is protected by the `oauth-example` security scheme, which uses the `clientCredentials` flow for token-based authentication.
+
+[id="serverless-logic-security-oauth-support-oidc-client-filter-extention_{context}"]
+== OAuth 2.0 Support with the OIDC Client filter extension
+
+OAuth 2.0 token management is handled by a Quarkus `OidcClient`. To enable this integration, you must add the Quarkus OIDC Client Filter extension to your project as shown in the following examples:
+
+.Example of adding extension using Maven
+[source,text,subs="attributes+"]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-oidc-client-filter</artifactId>
+  <version>{ServerlessLogicQuarkusVersion}</version>
+</dependency>
+
+<dependency>
+  <groupId>io.quarkiverse.openapi.generator</groupId>
+  <artifactId>quarkus-openapi-generator-oidc</artifactId>
+  <version>{ServerlessLogicOauthDependencyVersion}</version>
+</dependency>
+----
+
+.Example of adding extension using `gitops` profile
+
+Ensure that you configure the QUARKUS_EXTENSIONS build argument with the following value when building the workflow image:
+[source,text,subs="attributes+"]
+----
+$ --build-arg=QUARKUS_EXTENSIONS=io.quarkus:quarkus-oidc-client-filter:{ServerlessLogicQuarkusVersion},io.quarkiverse.openapi.generator:quarkus-openapi-generator-oidc:{ServerlessLogicOauthDependencyVersion}
+----
+
+[id="serverless-logic-security-oidc-configuration_{context}"]
+== `OidcClient` configuration
+
+To access the secured operation, define an `OidcClient` configuration in your `application.properties` file. The configuration uses the sanitized security scheme name from the OpenAPI specification, in this case, `oauth_example` as follows:
+
+[source,terminal]
+----
+quarkus.oidc-client.oauth_example.auth-server-url=https://example.com/oauth
+quarkus.oidc-client.oauth_example.token-path=/tokens
+quarkus.oidc-client.oauth_example.discovery-enabled=false
+quarkus.oidc-client.oauth_example.client-id=kogito-app
+quarkus.oidc-client.oauth_example.grant.type=client
+quarkus.oidc-client.oauth_example.credentials.client-secret.method=basic
+quarkus.oidc-client.oauth_example.credentials.client-secret.value=secret
+----
+
+In this configuration:
+
+* `oauth_example` matches the sanitized name of the `oauth-example` scheme in the OpenAPI file. The link between the sanitized scheme name and the corresponding `OidcClient` is by using that simple naming convention.
+
+* The OidcClient handles token generation and renewal automatically during workflow execution.
```

Comment:
```suggestion
* The OidcClient handles token generation and renewal automatically during workflow execution.

[NOTE]
====
For more information on these configurations you must refer to the Quakrus OpenID Connect and OAuth 2.0 documentation. 
====
```

Reply:
I think we can not mention the upstream docs like Quarkus OpenID Connect and OAuth 2.0 documentation in the prod docs